### PR TITLE
Feat(pt): Support fitting_net input statistics.

### PR DESCRIPTION
**Pull request by Chengqian-Zhang**
Solve issue #4281
Support fitting_net statistics to calculate the mean value and standard deviation of `fparam`/`aparam`. So that `fparam`/`aparam` can be normalized automatically before concatenating to descriptor.

## Summary by CodeRabbit

- **New Features**
	- Introduced a method to compute input statistics, including mean and standard deviation for fitting parameters.
	- Enhanced functionality to compute additional statistics alongside existing ones.
	- Added new parameters for data protection statistics to model configurations.
	- Added unit tests to validate the energy fitting model's statistical computations.

- **Bug Fixes**
	- Improved error handling for input data dimensions to ensure consistency.

### CODE REVIEW

File: deepmd/pt/model/task/fitting.py [hunk 1]
```diff
@@ -71,6 +75,82 @@ def share_params(self, base_class, shared_level, resume=False) -> None:
...
+                ss = torch.reshape(ss_, [-1, self.numb_aparam])
+                sys_sumv.append(torch.sum(ss, dim=0))
+                sys_sumv2.append(torch.sum(ss * ss, dim=0))
+                sys_sumn.append(ss.shape[0])
+            sumv = torch.sum(torch.stack(sys_sumv), dim=0)
+            sumv2 = torch.sum(torch.stack(sys_sumv2), dim=0)
+            sumn = sum(sys_sumn)
+            aparam_avg = sumv / sumn
+            aparam_std = torch.sqrt(sumv2 / sumn - (sumv / sumn) ** 2)
+            epsilon = 1e-12
```
Comment (wanghan-iapcm) [inline deepmd/pt/model/task/fitting.py:138]: The inverse epsilon would be too large.

File: source/tests/pt/test_fitting_stat.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+            all_data = np.array(tmp)
+        else:
+            all_data = np.concatenate((all_data, tmp), axis=0)
+    avg = np.average(all_data, axis=0)
+    std = np.std(all_data, axis=0)
+    return avg, std
+
+
+class TestEnerFittingStat(unittest.TestCase):
+    def test(self) -> None:
```
Comment (wanghan-iapcm) [inline source/tests/pt/test_fitting_stat.py:74]: please add a test case for near-zero std

File: deepmd/pt/model/task/fitting.py (hunk 1 above)
Comment (wanghan-iapcm) [reply deepmd/pt/model/task/fitting.py:138]: may be a better idea to make it a parameter and set default to e.g. 1e-3

**Review by wanghan-iapcm — CHANGES_REQUESTED** [review]

File: source/tests/pt/test_fitting_stat.py (hunk 2 above)
Comment (Chengqian-Zhang) [reply source/tests/pt/test_fitting_stat.py:74]: I have change `stds = [2, 0.4]` ---> `stds = [2, 0.4, 0.0001]`

File: deepmd/pt/model/task/fitting.py (hunk 1 above)
Comment (Chengqian-Zhang) [reply deepmd/pt/model/task/fitting.py]: I find that there is already a parameter `model/data_stat_protect` which has been used in tf backend. I use this parameter directly with default value=1e-2 in pt backend.

**Comment by Chengqian-Zhang** [comment]
Fail UT:
image

It looks like an occasional failure. Can you help me rerun the UT?

**Review by wanghan-iapcm — APPROVED** [review]

File: source/tests/pt/test_fitting_stat.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    merged_output_stat = []
+    nsys = len(sys_natoms)
+    ndof = len(avgs)
+    for ii in range(nsys):
+        sys_dict = {}
+        tmp_data_f = []
+        tmp_data_a = []
+        for jj in range(ndof):
+            tmp_data_f.append(
+                np.random.normal(  # noqa: NPY002
```
Comment (njzjz) [inline source/tests/pt/test_fitting_stat.py:28]: why ignore?

Comment (Chengqian-Zhang) [reply source/tests/pt/test_fitting_stat.py:29]: Ignore to solve warning in `pre-commit.ci`

Comment (Chengqian-Zhang) [reply source/tests/pt/test_fitting_stat.py:29]: I have set random seed now.

Comment (njzjz) [reply source/tests/pt/test_fitting_stat.py:29]: Please learn https://docs.astral.sh/ruff/rules/numpy-legacy-random/

**Comment by njzjz** [comment]
> I do not know why `docs/readthedocs.org` UT fails.

See https://github.com/breathe-doc/breathe/pull/1010

File: source/tests/pt/test_fitting_stat.py (hunk 3 above)
Comment (Chengqian-Zhang) [reply source/tests/pt/test_fitting_stat.py]: Thanks. I have deleted the ignore sentence.

**Review by njzjz — APPROVED** [review]

**Review by iProzd — APPROVED** [review]
